src/components/users/subscriptions/ko/runtime/subscriptions.ts

```diff
@@ -140,10 +142,12 @@ export class Subscriptions {
     private async isDelegationEnabled(subscriptionId: string): Promise<boolean> {
         const isDelegationEnabled = await this.tenantService.isSubscriptionDelegationEnabled();
         if (isDelegationEnabled) {
-            const delegation = new URLSearchParams();
-            delegation.append(DelegationParameters.SubscriptionId, Utils.getResourceName("subscriptions", subscriptionId));
-            this.router.navigateTo(`/${DelegationActionPath.unsubscribe}?${delegation.toString()}`);
-
+            const delegationParam = {};
+            delegationParam[DelegationParameters.SubscriptionId] =  Utils.getResourceName("subscriptions", subscriptionId);
+            const delegationUrl = await this.backendService.applyDelegation(DelegationAction.unsubscribe, delegationParam);
+            if (delegationUrl) {
+                await this.router.navigateTo(delegationUrl);
```

Comment:
Looks like you're doing redirect inside operation called `isDelegationEnabled`, which from its name sounds like it should only check for delegation. I suspect this will bring more confusion/bugs down the road. Maybe do redirect outside of after the check?

Reply:
Renamed function to `applyDelegation` to better understand what it is doing.
If we do redirect outside, we will repeatedly copy redirect code in many places.